kv_store/lww_kvs/kvs_ebs_server.cpp

```diff
@@ -829,12 +828,18 @@ int main(int argc, char* argv[]) {
       for (auto it = redistribution_map.begin(); it != redistribution_map.end(); it++) {
         zmq_util::send_msg((void*)it->second, &cache[it->first]);
       }
+
+      // delete the placement info for all the keys that are removed
+      for (auto it = key_to_query.begin(); it != key_to_query.end(); it++) {
+        // TODO: find a way to not remove the key as it may introduce race condition
```

Comment:
Not sure I follow this TODO. We're removing the key but shouldn't be?

Reply:
Yeah I didn't make this super-clear and it's sort of complicated...I'll explain it to you tmr